packages/cardpay-sdk/sdk/prepaid-card/base.ts

```diff
@@ -64,22 +64,36 @@ export default class PrepaidCard {
     return !hasBeenUsed && owner === issuer && owner !== ZERO_ADDRESS;
   }
 
+  async payMerchant(txnHash: string): Promise<TransactionReceipt>;
   async payMerchant(
     merchantSafe: string,
     prepaidCardAddress: string,
     spendAmount: number,
     txnOptions?: TransactionOptions,
     contractOptions?: ContractOptions
+  ): Promise<TransactionReceipt>;
+  async payMerchant(
+    merchantSafeOrTxnHash: string,
+    prepaidCardAddress?: string,
+    spendAmount?: number,
+    txnOptions?: TransactionOptions,
+    contractOptions?: ContractOptions
   ): Promise<TransactionReceipt> {
-    if (spendAmount < 50) {
+    if (!prepaidCardAddress) {
+      let txnHash = merchantSafeOrTxnHash;
+      return await waitUntilTransactionMined(this.layer2Web3, txnHash);
+    }
+    let merchantSafe = merchantSafeOrTxnHash;
+    let { nonce, onNonce, onTxnHash } = txnOptions ?? {};
+    if (spendAmount! < 50) {
```

Comment:
instead of putting a `!` after all the `spendAmount` reference, let's just add a test (I think this is what TS is nudging you to do):
```ts
if (spendAmount == null) {
  throw new Error(`spendAmount is not specified`);
}
```